clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2943,34 +2944,66 @@ void handleFailure(Throwable throwable) {
         return Collections.unmodifiableMap(this.metrics.metrics());
     }
 
+    @Deprecated
     @Override
     public ElectPreferredLeadersResult electPreferredLeaders(final Collection<TopicPartition> partitions,
                                                              ElectPreferredLeadersOptions options) {
-        final Set<TopicPartition> partitionSet = partitions != null ? new HashSet<>(partitions) : null;
+        final Set<TopicPartition> topicPartitions = partitions == null ? null : new HashSet<>(partitions);
+
         final KafkaFutureImpl<Map<TopicPartition, ApiError>> electionFuture = new KafkaFutureImpl<>();
         final long now = time.milliseconds();
-        runnable.call(new Call("electPreferredLeaders", calcDeadlineMs(now, options.timeoutMs()),
+
+        runnable.call(new Call("electLeaders", calcDeadlineMs(now, options.timeoutMs()),
```

Comment:
As far as I can tell, this call is the same as the one in `electLeaders` except the election type is statically defined. It's not a lot of code, but is it worth factoring it out?

Reply:
Yes. I'll revisit these set of changes and update the PR.